### PR TITLE
feat(DTFS2-7296): add to deal cancellation e2e tests

### PR DESCRIPTION
**Pull request by BethThomas141**
## Introduction :pencil2:
This adds the E2E tests for the bank request date and effective from date pages. 

image
image

## Resolution :heavy_check_mark:
List all changes made to the codebase.

## Miscellaneous :heavy_plus_sign:
List any additional fixes or improvements.

### CODE REVIEW

File: e2e-tests/e2e-fixtures/dateConstants.js [hunk 1]
```diff
@@ -134,6 +134,16 @@ const sixYearsOneDayDay = format(sixYearsOneDay, longDayFormat);
...
 
+const twelveMonthsOneDay = add(today, { months: 12, days: 1 });
+const twelveMonthsOneDayDay = format(twelveMonthsOneDay, longDayFormat);
+const twelveMonthsOneDayMonth = format(twelveMonthsOneDay, longMonthFormat);
+const twelveMonthsOneDayYear = format(twelveMonthsOneDay, longYearFormat);
+
+const twelveMonthsOneDayAgo = sub(today, { months: 12, days: 1 });
+const twelveMonthsOneDayAgoDay = format(twelveMonthsOneDayAgo, longDayFormat);
+const twelveMonthsOneDayAgoMonth = format(twelveMonthsOneDayAgo, longMonthFormat);
+const twelveMonthsOneDayAgoYear = format(twelveMonthsOneDayAgo, longYearFormat);
```
Comment (BethThomas141) [inline e2e-tests/e2e-fixtures/dateConstants.js:145]: I have a WIP PR to improve these date values - so for now I'm adding them in the existing format consistent with the other values in this file, but I'm going to prepose a refactor to this soon.

Comment (ttbarnes) [reply e2e-tests/e2e-fixtures/dateConstants.js:145]: @BethThomas141 Sounds good - can we touch base first though please, I have a WIP PR that improves consumption of the date constants https://github.com/UK-Export-Finance/dtfs2/pull/3556

File: e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/effective-from-date.spec.js [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+      cy.url().should('eq', relative(`/case/${dealId}/cancellation/cancel`));
+    });
+  });
+
+  describe('when logged in as a non-PIM user', () => {
+    beforeEach(() => {
+      cy.login(T1_USER_1);
+    });
+
+    it('should redirect when visiting effective from date page ', () => {
```
Comment (ttbarnes) [inline e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/effective-from-date.spec.js:110]: Minor, empty space at the end here 😛

**Review by ttbarnes — APPROVED** [review]
One very minor comment (not a blocker, can fix in next PR?) otherwise LGTM

File: e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/effective-from-date.spec.js (hunk 2 above)
Comment (BethThomas141) [reply e2e-tests/tfm/cypress/e2e/journeys/deal-cancellation/effective-from-date.spec.js:110]: Good spot, will fix in next PR as suggested :)

**Pull request merged**